modulus/metrics/general/crps.py

```diff
@@ -22,6 +22,43 @@
 Tensor = torch.Tensor
 
 
+@torch.jit.script
+def kcrps(pred: Tensor, obs: Tensor, dim: int = 0):
+    """
+    Computes the local Continuous Ranked Probability Score (CRPS) by using
+    the kernel version of CRPS
+
+    Creates a map of CRPS and does not accumulate over lat/lon regions.
+    Computes:
+        CRPS(X, y) = E[X - y] - 0.5 E[X-X']
+
+    Parameters
+    ----------
+    pred : Tensor
+        Tensor containing the ensemble predictions. The ensemble dimension
+        is assumed to be the leading dimension unless 'dim' is specified.
+    obs : Union[Tensor, np.ndarray]
+        Tensor or array containing an observation over which the CRPS is computed
+        with respect to.
+    dim : int, optional
+        The dimension over which to compute the CRPS, assumed to be 0.
+
+    Returns
+    -------
+    Tensor
+        Map of CRPS
+    """
+    pred = pred.unsqueeze(0).transpose(0, dim + 1).squeeze(dim + 1)
+    n = pred.shape[0]
+    _crps = 0.0 * obs
+    for i in range(n):
+        x_i = pred[i]
+        x_j = pred[i:]
+        _crps += torch.abs(x_i - obs) / n
```

Comment:
Nit: Can't this be done outside the for loop to coalesce a lot of these ops?

Reply:
To clarify, something like this outside this for loop where `_crps` is initialized
```python
_crps = torch.sum(torch.abs(pred - torch.unsqueeze(obs, 0)), dim=0)
```

Reply:
This is a good idea. I vectorized the previous code using vmap to eliminate the for loop.